yt_dlp/extractor/nfhsnetwork.py

```diff
@@ -0,0 +1,136 @@
+        # 'md5': '07708e7b6408384aa93df279db253fad',
+        'info_dict': {
+            'id': 'dd8de71d45',
+            'ext': 'mp4',
+            'title': '2015 UA Holiday Classic Tournament: National Division  - 12/26/2015',
+            # 'url': 'https://cfhls.nfhsnetwork.com/155125/155125.m3u8',
+            'description': '',
+            'uploader': 'SoCal Sports Productions',
+            'location': 'San Diego, California',
+            'timestamp': 1451187000,
+            'upload_date': '20151226'
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': True,
+        }
+    }, {
+        # Video with no broadcasts object
+        'url': 'https://www.nfhsnetwork.com/events/wiaa-wi/9aa2f92f82',
+        # 'md5': 'e9d5d94aa36f15ab7daf5a8488720c7f',
+        'info_dict': {
+            'id': '9aa2f92f82',
+            'ext': 'mp4',
+            'title': 'Competitive Equity  - 01/21/2015',
+            # 'url': 'https://cfhls.nfhsnetwork.com/121689/121689.m3u8',
+            'description': 'Committee members discuss points of their research regarding a competitive equity plan',
+            'uploader': 'WIAA - Wisconsin: Wisconsin Interscholastic Athletic Association',
+            'location': 'Stevens Point, Wisconsin',
+            'timestamp': 1421856000,
+            'upload_date': '20150121'
+        },
+        'params': {
+            # m3u8 download
+            'skip_download': True,
+        }
+    }
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage(url, video_id)
+        data = self._download_json(
+            'https://cfunity.nfhsnetwork.com/v2/game_or_event/' + video_id,
+            video_id)
+        publisher = data.get('publishers')[0]  # always exists
+        broadcast = (publisher.get('broadcasts') or publisher.get('vods'))[0]  # some (older) videos don't have a broadcasts object
+        uploader = publisher.get('formatted_name') or publisher.get('name') or ''
+        location = data.get('city') + ', ' + data.get('state_name')
+        description = broadcast.get('description') or ''
+        isLive = broadcast.get('on_air') or broadcast.get('status') == 'on_air' or False
+
+        timestamp = unified_timestamp(data.get('local_start_time'))
+        upload_date = unified_strdate(data.get('local_start_time'))
+
+        title = self._og_search_title(webpage) or self._html_search_regex(r'<h1 class="sr-hidden">(.*?)</h1>', webpage, 'title')
+        title = title[:title.find('|') - 1]
+
+        if broadcast.get('status') == 'complete':
+            m3u8_url = self._download_json('https://cfunity.nfhsnetwork.com/v2/vods/' + broadcast.get('key') + '/url', video_id)
+        else:
+            m3u8_url = self._download_json('https://cfunity.nfhsnetwork.com/v2/broadcasts/' + broadcast.get('key') + '/url', video_id)
+
+        m3u8_url = m3u8_url.get('video_url')
+
```

Comment:
can be simplified to something like:

```suggestion
        video_type = 'vods' if broadcast.get('status') == 'complete' else 'broadcasts'
        m3u8_url = self._download_json(
            'https://cfunity.nfhsnetwork.com/v2/%s/%s/url' % (video_type, broadcast.get('key')),
            video_id).get('video_url')
```
Also, why check for `complete` here, but `on_air` in `isLive`. Is there other possible values?

Reply:
I just realized I messed up the extractor right before opening the PR. Let me push a fix and try to incorporate your suggestion. 

[`54e2354` (#217)](https://github.com/yt-dlp/yt-dlp/pull/217/commits/54e23546b424f01baebee7e8204154085a8a13ae#diff-6d96ccb07fd17c8cfd0743ff2e5745605a5ab603574960ea2786000ea041394aR114)